### PR TITLE
Feat/dsdk 280: create context-module package for clear signing

### PR DESCRIPTION
**Pull request by aussedatlo**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** No impact yet
  - ...

### 📝 Description

Create the `context-module` package that will be used by the future `keyring-eth` to get all the context of a transaction using different backend.
This ticket is for:
- [DSDK-280](https://ledgerhq.atlassian.net/browse/DSDK-280): create context module
- [DSDK-281](https://ledgerhq.atlassian.net/browse/DSDK-281): create interfaces
- [DSDK-282](https://ledgerhq.atlassian.net/browse/DSDK-282): create entrypoint

### ❓ Context

- **JIRA or GitHub link**: [DSDK-280] [DSDK-281]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[DSDK-280]: https://ledgerhq.atlassian.net/browse/DSDK-280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DSDK-281]: https://ledgerhq.atlassian.net/browse/DSDK-281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DSDK-282]: https://ledgerhq.atlassian.net/browse/DSDK-282?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lambertkevin — DISMISSED** [review]
Just 2 minor things, other than that, all good 👍

File: libs/ledgerjs/packages/context-module/src/ContextModule.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ContextResponse } from "./models/ContextResponse";
+import { LoaderOptions } from "./models/LoaderOptions";
+import { Transaction } from "./models/Transaction";
+
+export interface ContextModule {
+  getContexts(transaction: Transaction, options: LoaderOptions): Promise<ContextResponse[]>;
```
Comment (mbertin-ledger) [inline libs/ledgerjs/packages/context-module/src/ContextModule.ts:6]: [COULD] rename as `resolveTransactionContexts`  ?

File: libs/ledgerjs/packages/context-module/src/DefaultContextModule.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { ContextModule } from "./ContextModule";
+import { ContextLoader } from "./loaders/ContextLoader";
+import { ContextResponse } from "./models/ContextResponse";
+import { LoaderOptions } from "./models/LoaderOptions";
+import { Transaction } from "./models/Transaction";
+
+type DefaultContextModuleConstructorArgs = {
+  loaders: ContextLoader[];
```
Comment (mbertin-ledger) [inline libs/ledgerjs/packages/context-module/src/DefaultContextModule.ts:8]: [COULD] Rename ContextResolver is maybe more meaningful that loader ? WDYT ?

File: libs/ledgerjs/packages/context-module/src/models/ContextResponse.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        | "provideNFTInformation"
+        | "provideDomainName"
+        | "setPlugin"
+        | "setExternalPlugin";
+      payload: string;
+    }
+  | {
+      type: "error";
+      error: Error;
+    };
```
Comment (mbertin-ledger) [inline libs/ledgerjs/packages/context-module/src/models/ContextResponse.ts:14]: [COULD] Declare Sub type  ? 

`type ContextSuccess = {
      type:
        | "provideERC20TokenInformation"
        | "provideNFTInformation"
        | "provideDomainName"
        | "setPlugin"
        | "setExternalPlugin";
      payload: string;
    }`
    
`export type ContextError = {
   ///... error details
    }`

**Review by mbertin-ledger — DISMISSED** [review]
👍 
Good Overral. 
My only comment is about the `loaders` naming, for me it's not very easy to understand the purpose, I would prefer using `resolvers`.

**Comment by aussedatlo** [comment]
Rebased with a new architecture, by scope

**Review by mbertin-ledger — APPROVED** [review]

**Review by lambertkevin — APPROVED** [review]

**Pull request merged**